bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/discovery/ZonePlayerDiscoveryParticipant.java

```diff
@@ -89,36 +89,31 @@ public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
                     case "ZP100":
                         modelName = "CONNECTAMP";
                         break;
-                    case "One SL":
-                        modelName = "OneSL";
-                        break;
-                    case "Arc SL":
-                        modelName = "ArcSL";
-                        break;
-                    case "Roam SL":
-                        modelName = "RoamSL";
-                        break;
                     case "Sub":
                         // The Sonos Sub is ignored
                         ignored = true;
                         break;
                     default:
+                        modelName = modelName.replaceAll("[^a-zA-Z0-9_]", "");
```

Comment:
Now that you are at it, two more ideas: Since` SonosXMLParser.extractModelName` now replaces ":" and " " in a similar fashion, maybe this logic could be entirely moved there instead of having three layers of model interpretation/translation?

And second, logic method `getModelName()` in this class also applies some logic regarding SYMFONISK. This could also be moved to `extractModelName()` and covered with tests.

Reply:
Done

Reply:
Excellent!